### PR TITLE
[Containerapp] Fix #29166: `az containerapp compose create`: Support image with tag and support get ACR from image

### LINKED ISSUE #29166: Can't specify image tag in compose when using `az containerapp compose create`
### Describe the bug

We are using the docker-compose file for container app creation using the command: `az containerapp compose create`. We have specified a tag in `image` attribute of service, we get the error: ```(InvalidImageString) Image name 'dev:latest:20240614142829041798' is invalid.```

I believe it is happening because [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py#L598) a timestamp is added as image tag even if there is already a tag. 

### Related command

`az containerapp compose create -g dms-rg-development --environment pr-test-env --compose-file-path "compose.yml" ----transport-mapping "db=tcp" --transport-mapping "redis=tcp" --transport-mapping "azurite=http"`

### Errors

(InvalidInputValue) Some of the properties of 'runRequest' are invalid.. InnerErrors: DockerBuildRequest:Image name 'dev:latest:20240614142829041798' is invalid. 
The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
Code: InvalidInputValue
Message: Some of the properties of 'runRequest' are invalid.. InnerErrors: DockerBuildRequest:Image name 'dev:latest:20240614142829041798' is invalid. 
The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
Target: request
Exception Details:      (InvalidImageString) Image name 'dev:latest:20240614142829041798' is invalid. 
        The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
        Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
        Code: InvalidImageString
        Message: Image name 'dev:latest:20240614142829041798' is invalid. 
        The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
        Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
        Target: DockerBuildRequest

### Issue script & Debug output

(InvalidInputValue) Some of the properties of 'runRequest' are invalid.. InnerErrors: DockerBuildRequest:Image name 'dev:latest:20240614142829041798' is invalid. 
The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
Code: InvalidInputValue
Message: Some of the properties of 'runRequest' are invalid.. InnerErrors: DockerBuildRequest:Image name 'dev:latest:20240614142829041798' is invalid. 
The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
Target: request
Exception Details:      (InvalidImageString) Image name 'dev:latest:20240614142829041798' is invalid. 
        The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
        Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
        Code: InvalidImageString
        Message: Image name 'dev:latest:20240614142829041798' is invalid. 
        The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. 
        Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
        Target: DockerBuildRequest

### Expected behavior

When an image tag is specified, no timestamp tag should be appended.

### Environment Summary

azure-cli                         2.60.0 *

core                              2.60.0 *
telemetry                          1.1.0

Extensions:
account                            0.2.5
containerapp                      0.3.50
rdbms-connect                      1.0.5

Dependencies:
msal                              1.28.0
azure-mgmt-resource             23.1.0b2

Python location '/opt/homebrew/Cellar/azure-cli/2.60.0/libexec/bin/python'
Extensions directory '/Users/workspace/.azure/cliextensions'

Python (Darwin) 3.11.9 (main, Apr  2 2024, 08:25:04) [Clang 15.0.0 (clang-1500.3.9.4)]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Greedygre**
**Related command**

`az containerapp compose create`

**Description**

Fix issue: https://github.com/Azure/azure-cli/issues/29166
Currently, when image in `--compose-file-path {path-to-compose.yml}` contains `tag`, it failed with error:
```
(InvalidInputValue) Some of the properties of 'runRequest' are invalid.. InnerErrors: DockerBuildRequest:Image name 'dev:latest:20240614142829041798' is invalid.
The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'.
Please refer to docs for more information on using image name and tag: https://aka.ms/az/acr/az-acr-build
Code: InvalidInputValue
Message: Some of the properties of 'runRequest' are invalid.. InnerErrors: DockerBuildRequest:Image name 'dev:latest:20240614142829041798' is invalid.
```

image

**This issue occurs because we always append a timestamp to the image, even when a tag already exists.**

Solution: **If a tag already exists in the image, we should not append a timestamp**.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Containerapp

File: src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py [hunk 1]
```diff
@@ -597,7 +597,16 @@ def run_acr_build(self, dockerfile, source, quiet=False, build_from_source=False
...
         tag_now_suffix = str(now).replace(" ", "").replace("-", "").replace(".", "").replace(":", "")
         image_name_with_tag = image_name + ":{}".format(tag_now_suffix)
-        self.image = self.registry_server + "/" + image_name_with_tag
+
+        if ":" in image_name.split("/")[-1]:
+            image_name_with_tag = image_name
+        else:
+            image_name_with_tag = image_name + ":{}".format(tag_now_suffix)
+
+        if not image_name_with_tag.startswith(self.registry_server):
```
Comment (LaylaLiu-gmail) [inline src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py:606]: any code to validate if the image_name_with_tag's server match the registry_server?

File: src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py [hunk 2]
```diff
@@ -597,7 +597,16 @@ def run_acr_build(self, dockerfile, source, quiet=False, build_from_source=False
         now = datetime.now()
         tag_now_suffix = str(now).replace(" ", "").replace("-", "").replace(".", "").replace(":", "")
         image_name_with_tag = image_name + ":{}".format(tag_now_suffix)
-        self.image = self.registry_server + "/" + image_name_with_tag
+
+        if ":" in image_name.split("/")[-1]:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py:601]: what about image name is a string "image"?

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py:601]: image

>>> image_name="xinyutest"
>>> print(image_name.split("/")[-1])
xinyutest
>>> print(":" in image_name.split("/")[-1])
False

File: src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py (hunk 1 above)
Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py:606]: image

Just add validation for this check.

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py:605]: Sync offline, not need to check.

Accepct:
2024/09/19 07:27:33 Successfully pushed image: xinyutest2.azurecr.io/caedd971c2c1acr.azurecr.io/k8se/quickstart:latest

**Review by LaylaLiu-gmail — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @Juliehzl 

Is this PR look good to you? Thanks for your reviewing.

**Review by Juliehzl — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**